tests/src/stateless_validator.rs

```diff
@@ -0,0 +1,114 @@
+#[cfg(test)]
```

Comment:
Nothing super interesting here -- mostly relying on stacked helpers to express test intentions.